### ISSUE TITLE
custom datetime formats with nullable<datetime>

### ISSUE THREAD

**Issue by ste-b** [agileobjects/AgileMapper#149]
custom datetime formats with nullable<datetime>
Hi Steve,
I have a case when in source class I have a nullable datetime property, while in target a property with same name is a string.
When executing the mapping, I obtain following exception:

	System.ArgumentException: Method 'System.String ToString(System.String)' declared on type 'System.DateTime' cannot be called with instance of type 'System.Nullable`1[System.DateTime]'

After some investigation, I realized that exception is probably thrown because I set a custom string format on datetime:

	WhenMapping
	  .StringsFrom<DateTime>(fs => fs.FormatUsing("dd/MM/yyyy"));

Instead, without custom format, mapping is successfull and I found the right value in target, of course without the format I need.
Can you give me some hint what is the right way to use the custom format, or maybe this case is not supported?
Thank you in advance,
ste

**Comment by SteveWilkes** [comment]
Hi!

There's a bug here which I'll fix - thanks for letting me know!

By default, the Mapper [maps DateTimes and nullable DateTimes to strings](https://dotnetfiddle.net/9HAd9q) using `CultureInfo.CurrentCulture.DateTimeFormat`, so if you can set that:

```C#
CultureInfo.CurrentCulture = CultureInfo.GetCultureInfo("en-GB");
```

...it should work. 

In the meantime I'll fix the bug! Thanks again!

Steve

**Comment by SteveWilkes** [comment]
Fixed in the latest release 1.5 branch code, release to follow. Thanks! :)

**Comment by ste-b** [comment]
Hi, thanks for your reply and for your hint
I also found a workaround for this, applying my custom convertion after mapping:

            WhenMapping
                .From<SrcDataType>().To<TgtDataType>()
                .After.Mapping(tgtData => tgtData.MyDate).Call((srcData, tgtData) =>
                    tgtData.MyDate = srcData.MyDate == null ? null : srcData.MyDate.Value.ToString("dd/MM/yyyy"));

Please let me know when this fix will be available in a nuget package, if possible.
Again, thanks a lot for your help :)
ste

**Comment by SteveWilkes** [comment]
This is fixed as of v1.5-preview3, which is now [on NuGet](https://www.nuget.org/packages/AgileObjects.AgileMapper).

Thanks again! :)
